sdk/python/kfp/components/_components.py

```diff
@@ -381,16 +381,24 @@ def component_default_to_func_default(component_default: str, is_optional: bool)
         ('input_paths', Mapping[str, str]),
         ('output_paths', Mapping[str, str]),
         ('inputs_consumed_by_value', Mapping[str, str]),
+        ('input_uris', Mapping[str, str]),
+        ('output_uris', Mapping[str, str]),
     ],
 )
 
 
+def _not_implemented(name: str) -> str:
```

Comment:
Instead of throwing `NotImplementedError` without too much information, can we explicitly say 'Input uri generator is not expected to be called when ...' and rename this function to make it clear that only a certain code path (when `InputUriPlaceholder` is used) will invoke this?

Reply:
It's not possible in general for the resolving function to generate the URI without knowing backend-specific context. So, resolving URI placeholders is only possible in compiler-specific, backend-specific bridges (e.g. Argo or IR) that override the generators using the information they have available.